pkg/util/util.go

```diff
@@ -1544,29 +1544,37 @@ func GetEnvWithDefault(key string, defaultval string) string {
 	return val
 }
 
+//IsInvalidKubeConfigError checks if specified error is due to invalid kubeconfig
+func IsInvalidKubeConfigError(err error) error {
+	if strings.Contains(err.Error(), "invalid configuration") {
+		return fmt.Errorf("invalid KUBECONFIG provided. Please point to a valid KUBECONFIG. You do not have to be logged in %w", err)
+	}
+	return nil
+}
+
 //IsValidKubeConfigPath checks if specified `KUBECONFIG` value is a valid file i.e the path should exist and be a file
 //if env `KUBECONFIG` is set then that is used or default `KUBECONFIG` is checked
-func IsValidKubeConfigPath() bool {
-	v := os.Getenv("KUBECONFIG")
-	if v == "" {
-		if home := homedir.HomeDir(); home != "" {
-			v = filepath.Join(home, ".kube", "config")
-			klog.V(4).Infof("using default kubeconfig path %s", v)
-		} else {
-			klog.V(4).Infof("no KUBECONFIG provided and cannot fallback to default")
-			return false
+func IsValidKubeConfigPath() error {
+	kubeConfigPath := os.Getenv("KUBECONFIG")
+	if kubeConfigPath != "" {
+		f1, err := os.Stat(kubeConfigPath)
+		if err != nil {
+			if os.IsNotExist(err) {
+				klog.V(4).Infof("invalid kubeconfig path set, KUBECONFIG env was set to %s which does no exist", kubeConfigPath)
+				return NewInvalidKubeConfigPathError()
+			}
+			return err
+		}
+		if f1.IsDir() {
+			klog.V(4).Infof("invalid kubeconfig path set, KUBECONFIG env was set to %s which is a directory", kubeConfigPath)
+			return NewInvalidKubeConfigPathError()
 		}
 	}
-	f1, err := os.Stat(v)
-	if os.IsNotExist(err) {
-		klog.V(4).Infof("invalid kubeconfig path set, KUBECONFIG env was set to %s which does no exist", v)
-		return false
-	}
-	if f1.IsDir() {
-		klog.V(4).Infof("invalid kubeconfig path set, KUBECONFIG env was set to %s which is a directory", v)
-		return false
-	}
-	return true
+	return nil
+}
+
+func NewInvalidKubeConfigPathError() error {
```

Comment:
Why `NewInvalidKubeConfigPathError` and not `InvalidKubeConfigPathError`? Also, why define a function instead of defining it as an error type? For example: https://github.com/openshift/odo/blob/089b8ea26c253592cd4788ed615a698297d620a5/pkg/kclient/operators.go#L15-L17

Reply:
Updating